vyper/functions/functions.py

```diff
@@ -1224,6 +1227,65 @@ def _can_compare_with_uint256(operand):
     )
 
 
+@signature('decimal')
+def sqrt(expr, args, kwargs, context):
+    from vyper.functions.utils import (
+        generate_inline_function,
+    )
+    arg = args[0]
+    sqrt_code = """
+assert x >= 0.0
+z: decimal
```

Comment:
So, this inline code injection is creating a variable? Would this variable be directly assigned (i.e. on every update in the inline code), or assigned at the end? `z` could be a memory or storage item.

Also, I think to be clearer, this mini-language inside our compiler should make it more explicitly clear what `x`, `y`, and `z` are, perhaps through some overhead or secondary syntax that shows they can be name-managled into working in case they shadow other variables. (it does do this, right?)

Reply:
It's not a mini language it's actual vyper. This isn't variable shadowing, the memory stays allocated no shadowing occurs, to rest of the compiler it just continues allocating. Nothing gets mangled the compiler is completely oblivous to this, and this is just another form of what happens a ton of places already just in LLL.

Reply:
Okay, my misunderstanding then. You know what my concern was :)